command/job/hunt.go

```diff
@@ -0,0 +1,165 @@
+package job
+
+import (
+	"fmt"
+	"github.com/hashicorp/nomad/api"
+	"strings"
+	"time"
+)
+
+func Hunt() error {
+	// create Nomad API client
+	nomadClient, err := api.NewClient(api.DefaultConfig())
+	if err != nil {
+		return err
+	}
+
+	jobs, _, err := nomadClient.Jobs().List(nil)
+	if err != nil {
+		return err
+	}
+	for _, job := range jobs {
+		if job.Type == "service" {
+			jAllocations, _, _ := nomadClient.Jobs().Allocations(job.ID, true, nil)
+			damn := false
+			for _, allocation := range jAllocations {
+				if allocation.JobVersion != jAllocations[0].JobVersion && allocation.ClientStatus == "running" {
+					damn = true
```

Comment:
why not just save an array of the mismatched allocations and check for array length to determine whether you need to print the discrepancies because you are looping through both twice and you only really need to print the ones that are off

Reply:
I added the array, and reworked the print,
but I do want to print all the allocations, versions and desired statuses